src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java

```diff
@@ -78,16 +78,19 @@ public UUID createNetworkCompositeModification(@NonNull List<UUID> modificationU
     }
 
     private void saveModificationsNonTransactional(UUID groupUuid, List<? extends ModificationEntity> modifications) {
+        int order = modificationRepository.countByGroupIdAndStashed(groupUuid, false);
         var modificationGroupEntity = this.modificationGroupRepository
             .findById(groupUuid)
             .orElseGet(() -> modificationGroupRepository.save(new ModificationGroupEntity(groupUuid)));
-        modifications.forEach(m -> {
+        for (ModificationEntity m : modifications) {
             modificationGroupEntity.addModification(m);
+            m.setModificationsOrder(order);
             // We need here to call the save() method on the modification entity cause the ids of the ModificationEntity's are used in further treatments in the same transaction.
             // As we generate the id in Java with @GeneratedValue(strategy = GenerationType.AUTO), saving the entity in the JPA world is enough to generate the id (no need to flush it).
             // Without the saving, the id generation would be done only at the flush() and wouldn't be available for the further treatments.
             modificationRepository.save(m);
-        });
+            order++;
+        }
```

Comment:
I'm not sure about this but I suggest to minimizing redundant save calls for example 

`for (ModificationEntity modification : modifications) {
            modification.setModificationsOrder(order);
            modificationGroupEntity.addModification(modification);  
            order++;
        }
        modificationGroupRepository.save(modificationGroupEntity);  `
        
and we have to set   CascadeType.PERSIST or CascadeType.ALL for the modifications in modificationGroupEntity

Reply:
maybe a saveAll at the end ?

Reply:
no because of the comment

Reply:
it's more efficient to save them all using saveAll(). This reduces the number of individual database insert/update operations and improving performance.